packages/mui-material/src/Grid/Grid.js

```diff
@@ -193,7 +193,16 @@ const GridRoot = styled('div', {
       container && styles.container,
       item && styles.item,
       zeroMinWidth && styles.zeroMinWidth,
-      container && spacing !== 0 && styles[`spacing-xs-${String(spacing)}`],
+      container &&
+        spacing &&
+        spacing !== 0 &&
+        typeof +spacing === 'number' &&
+        styles[`spacing-xs-${String(spacing)}`],
+      container && spacing?.xs && spacing.xs !== 0 && styles[`spacing-xs-${String(spacing.xs)}`],
```

Comment:
```suggestion
      getSpacingStyles(styles, container, 'xs);,
```

Same when extracting the classes below.

Would be great if we can simplify this to something like:

Reply:
Sure, good idea. I’ll work on simplifying.

Reply:
@mnajdova I added an additional commit that implements this idea.